scvi/dataset/_anndata.py

```diff
@@ -0,0 +1,508 @@
+    else:
+        logger.info("Generating sequential protein names")
+        protein_names = np.arange(adata.obsm[protein_expression_obsm_key].shape[1])
+
+    adata.uns["scvi_protein_names"] = protein_names
+
+    # batch mask totalVI
+    batch_mask = _get_batch_mask_protein_data(
+        adata, protein_expression_obsm_key, batch_key
+    )
+
+    # check if it's actually needed
+    if np.sum([~b for b in batch_mask]) > 0:
+        logger.info("Found batches with missing protein expression")
+        adata.uns["totalvi_batch_mask"] = batch_mask
+    return protein_expression_obsm_key
+
+
+def _setup_X(adata, X_layers_key):
+    # checking layers
+    if X_layers_key is not None:
+        assert (
+            X_layers_key in adata.layers.keys()
+        ), "{} is not a valid key in adata.layers".format(X_layers_key)
+        logger.info('Using data from adata.layers["{}"]'.format(X_layers_key))
+        X_loc = "layers"
+        X_key = X_layers_key
+        X = adata.layers[X_key]
+    else:
+        logger.info("Using data from adata.X")
+        X_loc = "_X"
+        X_key = None
+        X = adata._X
+
+    if _check_nonnegative_integers(X) is False:
+        logger_data_loc = (
+            "adata.X"
+            if X_layers_key is None
+            else "adata.layers[{}]".format(X_layers_key)
+        )
+        warnings.warn(
+            "{} does not contain unnormalized count data. Are you sure this is what you want?".format(
+                logger_data_loc
+            )
+        )
+    if adata.shape[0] / adata.shape[1] < 1:
+        warnings.warn("adata has more genes than cells. SCVI may not work properly.")
+
+    return X_loc, X_key
+
+
+def _setup_library_size(adata, batch_key, X_layers_key):
+    # computes the library size per batch
+    logger.info("Computing library size prior per batch")
+    local_l_mean_key = "_scvi_local_l_mean"
+    local_l_var_key = "_scvi_local_l_var"
+    _compute_library_size_batch(
+        adata,
+        batch_key=batch_key,
+        local_l_mean_key=local_l_mean_key,
+        local_l_var_key=local_l_var_key,
+        X_layers_key=X_layers_key,
+    )
+    return local_l_mean_key, local_l_var_key
+
+
+def _setup_summary_stats(adata, batch_key, labels_key, protein_expression_obsm_key):
+    n_batch = len(np.unique(adata.obs[batch_key]))
+    n_cells = adata.shape[0]
+    n_genes = adata.shape[1]
+    n_labels = len(np.unique(adata.obs[labels_key]))
+
+    summary_stats = {
+        "n_batch": n_batch,
+        "n_cells": n_cells,
+        "n_genes": n_genes,
+        "n_labels": n_labels,
+    }
+    if protein_expression_obsm_key is not None:
+        summary_stats["n_proteins"] = adata.obsm[protein_expression_obsm_key].shape[1]
+    adata.uns["scvi_summary_stats"] = summary_stats
+    logger.info(
+        "Successfully registered anndata object containing {} cells, {} genes, and {} batches.".format(
+            n_cells, n_genes, n_batch
+        )
+    )
```

Comment:
can we get the proteins in this info statement?

Reply:
done